File: libs/power-log-parser/src/lib/game-event-handler.ts
import { GameEvent } from './game-event';

export class GameEventHandler {
	onEvent: ((event: GameEvent) => void) | null = null;
	onEventsAll: ((events: GameEvent[]) => void) | null = null;

	private queuedEvents: GameEvent[] = [];

	Handle(gameEvent: GameEvent): void {
		if (this.queuedEvents.length > 0) {
			if (gameEvent != null) {
				this.queuedEvents.push(gameEvent);
			}
			console.debug('Sending queued events', '');
			if (this.onEventsAll) {
				this.onEventsAll(this.queuedEvents);
			}
			this.queuedEvents = [];
		} else {
			if (this.onEvent) {
				this.onEvent(gameEvent);
			}
		}
	}
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/GameEventHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace HearthstoneReplays
{
    public class GameEventHandler
    {
        public static Action<GameEvent> EventProvider;
        public static Action<IList<GameEvent>> EventProviderAll;

        private static IList<GameEvent> queuedEvents = new List<GameEvent>();

        public static void Handle(GameEvent gameEvent, bool isDevMode)
        {
            //if (queuedEvents.Count > 50 && lastEmitDate)
            // First event we get 
            if (queuedEvents.Count > 0)
            {
                if (gameEvent != null)
                {
                    queuedEvents.Add(gameEvent);
                }
                Logger.Log("Sending queued events", "");
                EventProviderAll?.Invoke(queuedEvents);
                queuedEvents.Clear();
            }
            else
            {
                //Logger.Log("Sending event to emit", gameEvent.Type);
                EventProvider?.Invoke(gameEvent);
            }
        }
    }
}